native-engine/datafusion-ext-commons/src/arrow/cast.rs

```diff
@@ -51,6 +51,11 @@ pub fn cast_impl(
             return try_cast_string_array_to_date(array);
         }
 
+        // use std parse to cast string to float
+        (&DataType::Utf8, to_dt) if to_dt.is_floating() => {
+            return try_cast_string_array_to_float(array, to_dt);
```

Comment:
i suggest adding a config to manually enable/disable this feature, since it has different behavior in different hive/spark versions.

Reply:
Added the conf